101/README.md

```diff
@@ -1,29 +1,81 @@
+This sample guides you through the full development flow. Follow this README to get familiar with the processes to setup your dev environment and build/deploy/validate sample codes using the Azure Arc VS code extension or command line options.
 
 ## Getting Started
 
-### Prerequisites
+### Recommended Dev Environment Setup
+- Visual Studio Code with [Azure Arc Extension](https://marketplace.visualstudio.com/search?term=azure%20arc&target=VSCode&category=All%20categories&sortBy=Relevance) from VS code marketpalce installed to simplify some development steps
+- Docker desktop or other containerization engine
+- A Kubernetes cluster. If you don't already have one, you can use the Azure Arc VS code extension to create an AKS EE cluster.
+
+### Building
+The same sample is implemented in 3 languages, build the corresponding Dockerfile to deploy to your K8S cluster.
+
+1. Make sure you are logged in to a container registry ready to use. If you're using docker desktop, please make sure you've logged in to your account either through the app UI or with docker commands and specify the password when prompted:
+    ```powershell
+    docker login docker.io/<your username> -u <your username>
+    ```
+
+2. Determine your image repository for the build process. If you're using docker.io as your container registry, the image repo should be in this format:
+    ```powershell
+    docker.io/<your username>/<your repo name>
+    ```
+
+    for example:
+
+    ```powershell
+    docker.io/johndoe/101sample
+    ```
+
+3. Build your docker image and push to a container registry
+    - If the Azure Arc VS code extension is installed, you can:
+        - Right click on the Dockerfile you selected and click on "Arc Extension: Build docker images" and specify your image repo. Or
+        - In VS Code, press "F1" or "Ctrl+Shift+P" to bring up the command palette and select "Arc Extension: Build docker images", specify your image repo and select the Dockerfile to build from the menu.
+    - If you prefer the command line option, use the following commands:
+    ```powershell
+    docker build -t <image repo>:<tag> -f <dockerfile path> <source code path>
+    docker push <image repo>:<tag>
+    ```
+
+    for example:
+    ```powershell
+    docker build -t docker.io/johndoe/101sample:latest -f .\101\dotnet\src\Dockerfile .\101\dotnet\src
+    docker push docker.io/johndoe/101sample:latest
+    ```
+
+### Deploying
+Once you have the image ready, you are ready to deploy the image to your K8S cluster.
+1. Make sure your K8S cluster to have sufficient privileges to pull images from it, e.g., you might need to provide imagePullSecrets in the values.yaml of the helm chart, or configure your container registry to allow anonymous pull.
+2. Fill out the variables in 101\101chart\values.yaml. The only mandatory field is the **repository** property under the **image** section, which should be the image repo + tag from the build step.
```

Comment:
repository property doesn't need tag. Tag should be put in tag property

Reply:
fixed.